File: algorithm/go/binarysearch/binary_search.go
package binarysearch

func BinarySearch(arr []int, elem int) int {
	low := 0
	high := len(arr) - 1

	for ;low <= high; {
		midIndex := (low + high) / 2
		guess := arr[midIndex]
		if guess == elem {
			return midIndex
		} else if guess < elem {
			low = midIndex + 1
		} else {
			high = midIndex - 1
		}
	}

	return -1
}

File: algorithm/js/binary_search.js
function BinarySearch(arr, elem) {
  var low = 0;
  var high = arr.length - 1;

  while(low !== high) {
    var midIndex = Math.floor((high-low)/2) + low;
    var mid = arr[midIndex]
    if (mid === elem) {
      return midIndex
    } else if (mid > elem) {
      high = midIndex - 1;
    } else {
      low = midIndex + 1
    }
  }

  return -1
}